crawl-ref/source/spl-book.cc

```diff
@@ -870,13 +870,10 @@ static bool _learn_spell_checks(spell_type specspell, bool wizard = false)
 */
 bool learn_spell(spell_type specspell, bool wizard)
 {
-    if (god_loathes_spell(specspell, you.religion))
-        mprf(MSGCH_WARN,"Your god extremely loathes this spell!");
-    else if (god_hates_spellcasting(you.religion))
-        mprf(MSGCH_WARN,"Your god hates spellcasting!");
-    else if (god_hates_spell(specspell, you.religion))
-        mprf(MSGCH_WARN,"Your god hates this spell!");
+    string mem_spell_warning_string = god_spell_warn_string(specspell, you.religion);
```

Comment:
I'd be inclined to just call it `warning`. That it's a string is implied by its type, and that it's related to memorising spells is implied from context. If you think it's important to emphasise that it's about memorising spells, I think you should not abbreviate to `mem`. Call it `memorisation_warning` or something.